tests/utils.py

```diff
@@ -0,0 +1,62 @@
+"""Utililty functions for the tests"""
+import logging
+import os
+from io import StringIO
+import unittest
+import pytest
+import warnings
+
+
+def verify_stand_alone_binary_parameter_updates(m):
```

Comment:
Would this be more useful if it took specific pairs - `("KIN", "KIN")` and additionally checked whether those were copied over? As it stands this passes if a parameter doesn't exist on either the PINT-facing or standalone side, so it's not actually checking whether such a parameter is copied over. Above you call this to check whether `KIN` is copied over but it doesn't actually test that.

Reply:
I agree this function can be improved.  Why don't we use another issue to discuss it? As it is, it tests the most copies and works for the KIN case, with two assertations in the test_ddk. 

ps: Can we remove the stand-alone binaries? I did not see any benefit, but a lot more risks and we need to test them.

Reply:
Well, you will need to coordinate with Patrick, as he's written code that depends on them working as they do now.

More, there's a place for using objects in python to handle multiple computations that share subexpressions without recomputing those subexpressions. One would create such an object from a model and a set of TOAs, these would be fixed, and subexpressions would be cached (say using `lazyproperty`). One might also be able to use modern Python tools to compute derivatives without implementing them analytically or doing numerical differentiation. So some kind of internal object, carefully designed, might be a good idea.

Reply:
Back to the topic of this PR. Do you think we have to change the function `verify_stand_alone_binary_parameter_updates(m)` to take specific pairs before merging this PR? I would say the current status of this test is sufficient for detecting the issue that is PR is aiming to fix.  I don't want to delay the 15 yr data analysis.

Reply:
I think the function is not as useful as it looks without the ability to check for the existence of specific pairs, but it's not a blocker for this PR necessarily.

Reply:
Let us fix this test function with the future development that is discussed in #1063. Maybe this function will not be needed any more.